src/com/palmergames/bukkit/towny/object/TownyObject.java

```diff
@@ -4,13 +4,16 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.HashSet;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public abstract class TownyObject implements Nameable {
 	private String name;
-
-	private HashSet<CustomDataField<?>> metadata = null;
+	
+	private Map<String, CustomDataField<?>> metadata = null;
```

Comment:
Even though when I originally did this I set this to `null` as the default, I think we're better of making it non-null:
```java
private Map<String, CustomDataField<?>> metadata = new HastMap<>();
```

Reply:
Reason I kept it null is for lazy-loading. I don't think the majority of TownyObjects will have metadata so I rather not create new objects for all of them if they're not going to be used.